src/plugins/intel_gna/src/gna_plugin.cpp

```diff
@@ -1231,58 +1270,56 @@ RequestStatus GNAPlugin::WaitFor(uint32_t request_idx, int64_t millisTimeout) {
     dnn->WriteInputAndOutputTextGNA(*worker.model());
 #endif
     for (auto&& outputBlobIt : requestResult) {
-        auto& outputBlob = outputBlobIt.second;
-        auto& outputDesc = outputs_.at(outputBlobIt.first);
-        if (outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::C &&
-            outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::NC &&
-            outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::CN &&
-            outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::NCHW &&
-            outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::CHW &&
-            outputBlob->getTensorDesc().getLayout() != InferenceEngine::Layout::SCALAR) {
+        const std::string& output_name = outputBlobIt.first;
+        Blob::Ptr output_blob = outputBlobIt.second;
+        const InferenceEngine::Layout& output_layout = output_blob->getTensorDesc().getLayout();
```

Comment:
getLayout() returns a value not a reference -
I would change `InferenceEngine::Layout&` to `InferenceEngine::Layout`

Reply:
fixed